File: DataSources/SqlServer/Settings.ascx.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2006
' by Perpetual Motion Interactive Systems Inc. ( http://www.perpetualmotion.ca )
'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'

Imports System.Collections.Generic
Imports System.IO

Imports DotNetNuke.Common.Utilities
Imports DotNetNuke.UI
Imports DotNetNuke.Services.Localization
Imports DotNetNuke.UI.Utilities
Imports DotNetNuke.Modules.Reports.Extensions

Namespace DotNetNuke.Modules.Reports.DataSources.SqlServer

    ''' -----------------------------------------------------------------------------
    ''' <summary>
    ''' The Settings class manages settings for the Sql Data Source
    ''' </summary>
    ''' <remarks>
    ''' </remarks>
    ''' <history>
    ''' </history>
    ''' -----------------------------------------------------------------------------
    Partial Public Class Settings
        Inherits ReportsSettingsBase
        Implements IDataSourceSettingsControl

        Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
            If Not ReportsClientAPI.IsSupported Then
                Me.AutomaticConnStrRadio.AutoPostBack = True
                Me.ManualConnStrRadio.AutoPostBack = True
                Me.IntegratedSecurityCheckBox.AutoPostBack = True

                AddHandler AutomaticConnStrRadio.CheckedChanged, AddressOf ConnStrModeCheckedChanged
                AddHandler ManualConnStrRadio.CheckedChanged, AddressOf ConnStrModeCheckedChanged
                AddHandler IntegratedSecurityCheckBox.CheckedChanged, AddressOf IntegratedSecurityCheckBox_CheckedChanged
            Else
                ReportsClientAPI.Import(Me.Page)
                ReportsClientAPI.ShowHideByRadioButtons(Me.Page, _
                                                        AutomaticConnStrRadio, _
                                                        ManualConnStrRadio, _
                                                        AutomaticConnStrConfig, _
                                                        ManualConnStrConfig)
                ReportsClientAPI.ShowHideByCheckBox(Me.Page, _
                                                    IntegratedSecurityCheckBox, _
                                                    UserNameRow, _
                                                    False)
                ReportsClientAPI.ShowHideByCheckBox(Me.Page, _
                                                    IntegratedSecurityCheckBox, _
                                                    PasswordRow, _
                                                    False)

                ' Show the correct table
                UpdateControlVisibility()
            End If
        End Sub

        Public Overrides Sub LoadSettings(ByVal Settings As System.Collections.Generic.Dictionary(Of String, String))
            Dim useConnectionString As Boolean = SettingsUtil.GetDictionarySetting(Of Boolean)(Settings, ReportsController.SETTING_UseConnectionString, False)
            Me.AutomaticConnStrRadio.Checked = Not useConnectionString
            Me.ManualConnStrRadio.Checked = useConnectionString
            ConnStrTextBox.Text = SettingsUtil.GetDictionarySetting(Of String)(Settings, ReportsController.SETTING_ConnectionString, String.Empty)
            ServerTextBox.Text = SettingsUtil.GetDictionarySetting(Of String)(Settings, ReportsController.SETTING_Server, String.Empty)
            DatabaseTextBox.Text = SettingsUtil.GetDictionarySetting(Of String)(Settings, ReportsController.SETTING_Database, String.Empty)
            IntegratedSecurityCheckBox.Checked = SettingsUtil.GetDictionarySetting(Of Boolean)(Settings, ReportsController.SETTING_Sql_UseIntegratedSecurity, False)
            If Not IntegratedSecurityCheckBox.Checked Then
                UserNameTextBox.Text = SettingsUtil.GetDictionarySetting(Of String)(Settings, ReportsController.SETTING_UserName, String.Empty)
                PasswordTextBox.Text = SettingsUtil.GetDictionarySetting(Of String)(Settings, ReportsController.SETTING_Password, String.Empty)
            End If
            SqlDataSourceCommonSettingsControl.LoadSettings(Settings)

            UpdateControlVisibility()
        End Sub

        Public Overrides Sub SaveSettings(ByVal Settings As System.Collections.Generic.Dictionary(Of String, String))
            Settings.Clear()
            If Me.AutomaticConnStrRadio.Checked Then
                Settings.Add(ReportsController.SETTING_UseConnectionString, "False")
                Settings.Add(ReportsController.SETTING_Server, ServerTextBox.Text)
                Settings.Add(ReportsController.SETTING_Database, DatabaseTextBox.Text)
                Settings.Add(ReportsController.SETTING_Sql_UseIntegratedSecurity, IntegratedSecurityCheckBox.Checked.ToString())
                If Not IntegratedSecurityCheckBox.Checked Then
                    Settings.Add(ReportsController.SETTING_UserName, UserNameTextBox.Text)
                    Settings.Add(ReportsController.SETTING_Password, PasswordTextBox.Text)
                End If
            ElseIf Me.ManualConnStrRadio.Checked Then
                Settings.Add(ReportsController.SETTING_UseConnectionString, "True")
                Settings.Add(ReportsController.SETTING_ConnectionString, ConnStrTextBox.Text)
            End If
            SqlDataSourceCommonSettingsControl.SaveSettings(Settings)
        End Sub

        Private Sub ConnStrModeCheckedChanged(ByVal sender As Object, ByVal args As EventArgs)
            UpdateControlVisibility()
        End Sub

        Private Sub IntegratedSecurityCheckBox_CheckedChanged(ByVal sender As Object, ByVal args As EventArgs)
            UpdateControlVisibility()
        End Sub

        Private Sub UpdateControlVisibility()
            If Me.AutomaticConnStrRadio.Checked Then
                ReportsClientAPI.ClientShow(Me.AutomaticConnStrConfig)
                ReportsClientAPI.ClientHide(Me.ManualConnStrConfig)
            ElseIf Me.ManualConnStrRadio.Checked Then
                ReportsClientAPI.ClientShow(Me.ManualConnStrConfig)
                ReportsClientAPI.ClientHide(Me.AutomaticConnStrConfig)
            End If

            If Me.IntegratedSecurityCheckBox.Checked Then
                ReportsClientAPI.ClientHide(Me.UserNameRow)
                ReportsClientAPI.ClientHide(Me.PasswordRow)
            Else
                ReportsClientAPI.ClientShow(Me.UserNameRow)
                ReportsClientAPI.ClientShow(Me.PasswordRow)
            End If
        End Sub

        Public ReadOnly Property DataSourceClass() As String Implements IDataSourceSettingsControl.DataSourceClass
            Get
                Return GetType(SqlServerDataSource).FullName
            End Get
        End Property
    End Class

End Namespace

File: DataSources/SqlServer/Settings.ascx.cs
#region Copyright
// 
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2018
// by DotNetNuke Corporation
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
//
#endregion


namespace DotNetNuke.Modules.Reports.DataSources.SqlServer
{
    using System;
    using System.Collections.Generic;
    using global::DotNetNuke.Modules.Reports.Extensions;

    /// -----------------------------------------------------------------------------
    /// <summary>
    ///     The Settings class manages settings for the Sql Data Source
    /// </summary>
    /// <remarks>
    /// </remarks>
    /// <history>
    /// </history>
    /// -----------------------------------------------------------------------------
    public partial class Settings : ReportsSettingsBase, IDataSourceSettingsControl
    {
        public override void LoadSettings(Dictionary<string, string> Settings)
        {
            var useConnectionString =
                Convert.ToBoolean(
                    SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_UseConnectionString, false));
            this.AutomaticConnStrRadio.Checked = !useConnectionString;
            this.ManualConnStrRadio.Checked = useConnectionString;
            this.ConnStrTextBox.Text =
                Convert.ToString(
                    SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_ConnectionString,
                                                      string.Empty));
            this.ServerTextBox.Text =
                Convert.ToString(
                    SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_Server, string.Empty));
            this.DatabaseTextBox.Text =
                Convert.ToString(
                    SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_Database, string.Empty));
            this.IntegratedSecurityCheckBox.Checked =
                Convert.ToBoolean(
                    SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_Sql_UseIntegratedSecurity,
                                                      false));
            if (!this.IntegratedSecurityCheckBox.Checked)
            {
                this.UserNameTextBox.Text =
                    Convert.ToString(
                        SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_UserName, string.Empty));
                this.PasswordTextBox.Text =
                    Convert.ToString(
                        SettingsUtil.GetDictionarySetting(Settings, ReportsController.SETTING_Password, string.Empty));
            }
            this.SqlDataSourceCommonSettingsControl.LoadSettings(Settings);

            this.UpdateControlVisibility();
        }

        public override void SaveSettings(Dictionary<string, string> Settings)
        {
            Settings.Clear();
            if (this.AutomaticConnStrRadio.Checked)
            {
                Settings.Add(ReportsController.SETTING_UseConnectionString, "False");
                Settings.Add(ReportsController.SETTING_Server, this.ServerTextBox.Text);
                Settings.Add(ReportsController.SETTING_Database, this.DatabaseTextBox.Text);
                Settings.Add(ReportsController.SETTING_Sql_UseIntegratedSecurity,
                             this.IntegratedSecurityCheckBox.Checked.ToString());
                if (!this.IntegratedSecurityCheckBox.Checked)
                {
                    Settings.Add(ReportsController.SETTING_UserName, this.UserNameTextBox.Text);
                    Settings.Add(ReportsController.SETTING_Password, this.PasswordTextBox.Text);
                }
            }
            else if (this.ManualConnStrRadio.Checked)
            {
                Settings.Add(ReportsController.SETTING_UseConnectionString, "True");
                Settings.Add(ReportsController.SETTING_ConnectionString, this.ConnStrTextBox.Text);
            }
            this.SqlDataSourceCommonSettingsControl.SaveSettings(Settings);
        }

        public string DataSourceClass => typeof(SqlServerDataSource).FullName;

        protected void Page_Load(object sender, EventArgs e)
        {
            if (!ReportsClientAPI.IsSupported)
            {
                this.AutomaticConnStrRadio.AutoPostBack = true;
                this.ManualConnStrRadio.AutoPostBack = true;
                this.IntegratedSecurityCheckBox.AutoPostBack = true;

                this.AutomaticConnStrRadio.CheckedChanged += this.ConnStrModeCheckedChanged;
                this.ManualConnStrRadio.CheckedChanged += this.ConnStrModeCheckedChanged;
                this.IntegratedSecurityCheckBox.CheckedChanged += this.IntegratedSecurityCheckBox_CheckedChanged;
            }
            else
            {
                ReportsClientAPI.Import(this.Page);
                ReportsClientAPI.ShowHideByRadioButtons(this.Page, this.AutomaticConnStrRadio, this.ManualConnStrRadio,
                                                        this.AutomaticConnStrConfig, this.ManualConnStrConfig);
                ReportsClientAPI.ShowHideByCheckBox(this.Page, this.IntegratedSecurityCheckBox, this.UserNameRow,
                                                    false);
                ReportsClientAPI.ShowHideByCheckBox(this.Page, this.IntegratedSecurityCheckBox, this.PasswordRow,
                                                    false);

                // Show the correct table
                this.UpdateControlVisibility();
            }
        }

        private void ConnStrModeCheckedChanged(object sender, EventArgs args)
        {
            this.UpdateControlVisibility();
        }

        private void IntegratedSecurityCheckBox_CheckedChanged(object sender, EventArgs args)
        {
            this.UpdateControlVisibility();
        }

        private void UpdateControlVisibility()
        {
            if (this.AutomaticConnStrRadio.Checked)
            {
                ReportsClientAPI.ClientShow(this.AutomaticConnStrConfig);
                ReportsClientAPI.ClientHide(this.ManualConnStrConfig);
            }
            else if (this.ManualConnStrRadio.Checked)
            {
                ReportsClientAPI.ClientShow(this.ManualConnStrConfig);
                ReportsClientAPI.ClientHide(this.AutomaticConnStrConfig);
            }

            if (this.IntegratedSecurityCheckBox.Checked)
            {
                ReportsClientAPI.ClientHide(this.UserNameRow);
                ReportsClientAPI.ClientHide(this.PasswordRow);
            }
            else
            {
                ReportsClientAPI.ClientShow(this.UserNameRow);
                ReportsClientAPI.ClientShow(this.PasswordRow);
            }
        }
    }
}